enhancements/windows-containers/windows-machine-config-operator-upgrades.md

```diff
@@ -0,0 +1,188 @@
+---
+title: windows-machine-config-operator-upgrades
+authors:
+  - "@sebsoto"
+reviewers:
+  - "@sdodson"
+  - "@derekwaynecarr"
+approvers:
+  - "@sdodson"
+  - "@derekwaynecarr"
+creation-date: 2020-06-24
+last-updated: 2020-06-24
+status: implementable
+---
+
+# Windows Machine Config Operator Upgrades
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement allows a cluster administrator to upgrade a deployed Windows 
+Machine Config Operator (WMCO), and associated Windows node components, to a
+newer version.
+
+## Motivation
+
+The main motivation behind this enhancement is to give the end user
+an easy and non-disruptive way to upgrade to a newer release of the WMCO, and
+upgrade associated Windows node components with it.
+WMCO is not part of the release payload, and is managed with [OLM](https://github.com/operator-framework/operator-lifecycle-manager).
+
+### Goals
+
+As part of this enhancement we plan to do the following:
+* Enable upgrading WMCO with as little disruption as possible.
+* Enable upgrading on-node Kubernetes components through newer versions of the
+  WMCO, including major version upgrades across OpenShift cluster upgrades, and
+  bug and CVE fixes during the current version.
+
+### Non-Goals
+
+In this enhancement we do not plan to support Windows operating system updates.
+
+## Proposal
+
+To allow for the WMCO to be upgraded in a safe and non-disruptive way, it is
+proposed that Windows VMs configured by previous versions of the WMCO are
+indirectly drained and terminated in batches by user created MachineHealthChecks,
```

Comment:
Have we tested this functionality manually? I gave this suggestion as it is available in the documentation. I have not tested it manually or we need to check with cloud team on this.

Reply:
I have not manually tested this functionality. I have looked through the MachineHealthCheck controller code and it looks like as long as the `nodeRef` field is populated accordingly there should be no difference than if this was a linux node.

Reply:
https://github.com/openshift/machine-api-operator/blob/168f5015e9b88dc9dbcdee1e9f6180c766470e9a/pkg/controller/machinehealthcheck/machinehealthcheck_controller.go#L312

Reply:
Yup, I saw that too, I think it would be better if we get some sort of confirmation from the cloud team if  it is supported.